components/CameraComponent.tsx

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Camera } from 'expo-camera';
+
+export default function CameraComponent({ flashMode }: { flashMode: string }) {
+  if (flashMode) {
+    return <Camera type={Camera.Constants.Type.back} flashMode={flashMode} />;
```

Comment:
I think you could remove this `if-check`, always render `Camera` and just pass either `torch` or `off` (and update `TORCH_OFF` in `NativeModule` to be `off`, not empty string).

You can then change `flashMode` type above (in props) to `flashMode: 'torch' | 'off'`